architecture.md

```diff
@@ -0,0 +1,91 @@
+
+## Communication process
+
+Any information that crosses the process boundary (whether back-to-front with scene broadcaster or front-to-back with user commands) has to go through [Ignition Transport](https://ignitionrobotics.org/libs/transport), the communication library.
+Synchronization between processes is performed by scene broadcaster, a server-side Ignition Gazebo plugin that uses Ignition Transport and [Ignition Messages](https://ignitionrobotics.org/libs/msgs) libraries to send messages from the server to the client.
+The messages themselves are provided by Ignition Messages, while the framework for creating the publishers and subscribers that exchange messages is from Ignition Transport.
+
+The scene broadcaster system is responsible for getting the state of the world (all of the entities and their component values) from the ever-changing simulation loop running in the back end, packaging that information into a very compact message, and periodically sending it to the frontend to process.
+Omitting the scene broadcaster, while possible, would mean not being able to visualize data on the frontend, which can be useful for saving computational power.
+
+In addition to state messages sent from the scene broadcaster to the client, the client can also make requests to the server using the user commands system (another backend plugin).
+It also utilizes the Ignition Transport and Ignition Messages libraries.
+The user command system has services that can be requested by the frontend GUI to insert, delete, move, etc., and will send responses back to the GUI acknowledging the receipt and execution of those requests.
+
+Several non-default plugins are able to send or receive messages to other processes outside of the server-client process.
+For example, the server-side sensors plugin publishes messages outside of the server-client process, and the server-side diff-drive plugin receives info from processes other than the client.
+One such process these plugins may communicate with, for example, is ROS.
+
+## Frontend client process
+
+The client-side process, essentially the GUI, also comprises multiple plugins, some loaded by default, others that can be added, and all optional.
+All of the frontend plugins use the [Ignition GUI](https://ignitionrobotics.org/libs/gui) library.
+The client itself also relies on Ignition GUI.
+There are visualization plugins that create the windows of the GUI, add buttons and other interactive features, and a 3D scene plugin that utilizes [Ignition Rendering](https://ignitionrobotics.org/libs/rendering) that creates the scene the user sees.
+
+Frontend plugins communicate using events.
+They all have access to the entity and component information from the compressed message provided by the scene broadcaster system from the backend.
+The Entity Component Manager on the client side performs the actual message unpacking and distributes the entity and component information to the relevant plugins.
```

Comment:
In the diagram, we have an arrow that goes from server to client labelled `packed messages`, which points directly to the `unpacked message contents` ECM on the client side. Should we add something to the diagram on the client side that describes this message unpacking process? @chapulina feel free to chime in with your thoughts

Reply:
humm I'm not sure how to add this to the diagram itself. I'm also not sure what other information to provide that can make the process clearer...